### PR TITLE
DetectLanguaje Client EsES/EsMX/

### PR DESCRIPTION
**Pull request by ErnestoGonzalezC**
## Changes Proposed:
-  detects the client's language. 

## Issues Addressed:
- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

When I was working with the emulator, I wanted to use modules to try to send different menus depending on the client's language.

How to use ?

Simply use the if conditional to indicate the verification of whether the client is in Spanish and else if it is any other language.

Example:
```
bool OnGossipHello(Player * player, Creature * creature)
{
    player->PlayerTalkClass->ClearMenus();


    if (player->ClientInSpanishOrMexican())
    {
	player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "S7 JcJ Gear", GOSSIP_SENDER_MAIN, 1);
	player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "S7 jcE Gear", GOSSIP_SENDER_MAIN, 2);
    }
    else
    {
	player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "T7 PvP Gear", GOSSIP_SENDER_MAIN, 1);
	player->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "T7 PvE Gear", GOSSIP_SENDER_MAIN, 2);
    }

    player->PlayerTalkClass->SendGossipMenu(1, creature->GetGUID());
    return true;
}
```

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.h [hunk 1]
```diff
@@ -1115,6 +1115,15 @@ class Player : public Unit, public GridObject<Player>
...
     }
+	
+	// Detect Client Languaje
+    bool ClientInSpanishOrMexican()
+    {
+        if (GetSession()->GetSessionDbLocaleIndex() == LOCALE_esES || GetSession()->GetSessionDbLocaleIndex() == LOCALE_esMX)
+            return true;
+        else
+            return false;
+    }
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.h:1126]: where is this method supposed to be used?

Comment (ErnestoGonzalezC) [reply src/server/game/Entities/Player/Player.h:1126]: @FrancescoBorzi the idea is to use it for the greater scope of the modules.

Comment (ErnestoGonzalezC) [reply src/server/game/Entities/Player/Player.h:1126]: Personally, it has been quite useful to be able to offer menus in both Spanish and English

Comment (ErnestoGonzalezC) [reply src/server/game/Entities/Player/Player.h:1126]: For example in the TemplateNPC module.
```
           if (sTemplateNpcMgr->CanEquipTemplate(player, playerSpecStr) == false)
            {
                if (player->ClientInSpanishOrMexican())
                {
                    player->GetSession()->SendAreaTriggerMessage("No hay plantillas para esta especializacion %s .", playerSpecStr.c_str());
                }
                else
                {
                    player->GetSession()->SendAreaTriggerMessage("There's no templates for %s specialization yet.", playerSpecStr.c_str());
                }
                
                return;
```

**Comment by Winfidonarleyan** [comment]
Why not use?
```cpp
player->GetSession()->GetSessionDbLocaleIndex() == LOCALE_esES || player->GetSession()->GetSessionDbLocaleIndex() == LOCALE_esMX
```

**Comment by FrancescoBorzi** [comment]
@ErnestoGonzalezC what about the comment from @Winfidonarleyan  ? isn't that also an option?

**Comment by ErnestoGonzalezC** [comment]
@FrancescoBorzi  Yes it is, although a bit more uncomfortable to use in my opinion.

**Comment by ErnestoGonzalezC** [comment]
If you consider that it is something unnecessary I will understand.

File: src/server/game/Entities/Player/Player.h (hunk 1 above)
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.h:1126]:
```suggestion
    bool isSpanishClient()
    {
        return GetSession()->GetSessionDbLocaleIndex() == LOCALE_esES || GetSession()->GetSessionDbLocaleIndex() == LOCALE_esMX;
    }
```

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
@ErnestoGonzalezC I've renamed it to `hasSpanishClient` and added `[[nodiscard]]` , thanks for your contribution!

**Comment by yuanf225** [comment]
Can support more languages？

**Comment by ErnestoGonzalezC** [comment]
> Can support more languages？

I could create a variable for each of these languages.
```
enum LocaleConstant
{
    LOCALE_enUS = 0,
    LOCALE_koKR = 1,
    LOCALE_frFR = 2,
    LOCALE_deDE = 3,
    LOCALE_zhCN = 4,
    LOCALE_zhTW = 5,
    LOCALE_esES = 6,
    LOCALE_esMX = 7,
    LOCALE_ruRU = 8
};
```

**Comment by yuanf225** [comment]
nico

**Pull request merged**